library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java

```diff
@@ -106,4 +106,20 @@ public long getId() {
     public void setId(long id) {
         this.mId = id;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        WeekViewEvent that = (WeekViewEvent) o;
+
+        return mId == that.mId;
+
+    }
+
+    @Override
+    public int hashCode() {
+        return (int) (mId ^ (mId >>> 32));
+    }
```

Comment:
I actually implicit use the `equals` and `hashCode`  in the `MonthChangeListener` when we feed data in.  we need async call our server to get event data, and feed in. Unfortunately, the onMonthChange are called sync way.  So we make a pain implement to make it happen. We Use Set to make sure same event only add once into the Set.  So the WeekView can display correctly.  That's not good, but just work for me.

```
// cache set of event
private Set<WeekViewEvent> events=null;

// async call our server to get event data
WeekView.MonthChangeListener mMonthChangeListener = new WeekView.MonthChangeListener() {
        public List<WeekViewEvent> onMonthChange(int newYear, int newMonth) {
        // ... call server asnyc
        
RequstServer.getInstance(this).getEvent(request, new Succeed<Response>() {
            @Override
            public void onSucced(Response object) {
                
                    for(int i=0;i<object.getList().size();i++){
                        WeekViewEvent event=new WeekViewEvent();
                        
                        // ...... populate event and add to Set.
                        
                        events.add(event);
                    }
                    weekView.notifyDatasetChanged();
                }
            }
        });

        ArrayList<WeekViewEvent> evs = new ArrayList<>();
            evs.addAll(events);
            return evs;
     }
    };
```